test/integration_test_helper.rb

```diff
@@ -20,18 +20,63 @@
 end
 
 Selenium::WebDriver::Chrome::Service.driver_path = ENV['TESTDRIVER_PATH'] || Foreman::Util.which('chromedriver', Rails.root.join('node_modules', '.bin'))
-Capybara.register_driver :selenium_chrome do |app|
+
+javascript_driver = ENV.fetch("JS_TEST_DRIVER") { ENV['DEBUG_JS_TEST'] ? :selenium_chrome : :selenium_chrome_headless }.to_sym
+
+def get_chrome_options
   options = Selenium::WebDriver::Chrome::Options.new
-  options.args << '--disable-gpu'
-  options.args << '--no-sandbox'
   options.args << '--window-size=1024,768'
-  options.args << '--headless' unless ENV['DEBUG_JS_TEST'] == '1'
-  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
+  options.args += ENV.fetch('ADDITIONAL_CHROME_OPTIONS', '').split(';')
+  options
+end
+
+if javascript_driver == :selenium_chrome_remote
+  ShowMeTheCookies.register_adapter(:selenium_chrome_remote, ShowMeTheCookies::SeleniumChrome)
+
+  Capybara.register_driver :selenium_chrome_remote do |app|
+    selenium_remote_host = ENV.fetch('SELENIUM_REMOTE_HOST', nil)
+    selenium_remote_port = ENV.fetch('SELENIUM_REMOTE_PORT', 4444)
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :remote,
+      url: "http://#{selenium_remote_host}:#{selenium_remote_port}/wd/hub",
+      options: get_chrome_options)
+  end
+elsif javascript_driver == :selenium_chrome_headless
+  options = get_chrome_options
+  options.args << '--headless'
+  Capybara.register_driver :selenium_chrome_headless do |app|
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :chrome,
+      options: options)
+  end
+else
+  Capybara.register_driver javascript_driver do |app|
+    Capybara::Selenium::Driver.new(
+      app,
+      browser: :chrome,
+      options: get_chrome_options)
+  end
 end
+
 Capybara.configure do |config|
-  config.javascript_driver      = ENV["JS_TEST_DRIVER"]&.to_sym || :selenium_chrome
-  config.default_max_wait_time  = 20
+  config.javascript_driver = javascript_driver
+  config.default_max_wait_time = 20
   config.enable_aria_label = true
+  if ENV.fetch("JS_TEST_DRIVER", nil) == 'selenium_chrome_remote'
+    app_host = ENV.fetch('APP_SERVER_HOST') do
+      Socket.ip_address_list
+        .find(&:ipv4_private?)
+        .ip_address
+    end
+    app_port = ENV.fetch('APP_SERVER_PORT', "8080")
+    config.server_port = app_port
+    # application server
+    config.server_host = "0.0.0.0"
+    # address used by selenium host to connect to application server
+    config.app_host = "http://#{app_host}:#{app_port}"
```

Comment:
From reading the code I'd think this is redundant if you already set `server_host` correctly. In fact, it will even understand HTTP vs HTTPS then. It would also avoid the whole determining a port issue I described above.

At least, when I read:

https://github.com/teamcapybara/capybara/blob/52eaecea6d154b7d664b0032cd1cbcad4788fe65/lib/capybara/session.rb#L266

And `server_url` is defined by:

https://github.com/teamcapybara/capybara/blob/52eaecea6d154b7d664b0032cd1cbcad4788fe65/lib/capybara/session.rb#L837-L839

And `base_url` is defined by:

https://github.com/teamcapybara/capybara/blob/52eaecea6d154b7d664b0032cd1cbcad4788fe65/lib/capybara/server.rb#L91-L93

Reply:
![foreman_selenium drawio](https://github.com/theforeman/foreman/assets/2125774/d13df8f2-4388-4f3a-bb58-6b09be8948f9)

I am also not entirely happy with this solution but it is the only way I could find to get it to work (without changes to capybara).

When app_host is not set capypara tries to access the IP set by `config.server_host`:
```
http://10.11.12.1:4444/wd/hub/session/a9f8f81dc07d3ccebc426e537e22d29a/url | {"url":"http://0.0.0.0:46703/users/login"}
```
But the docker container running outside of the vagrant box needs the IP of the vagrant box (see picture)

Sadly capybara does not allow to just change the IP but requires the workaround by `app_host` which includes scheme, ip, and port.